### PR TITLE
Alert Dialog when deleting flight

### LINKED ISSUE #239: Add confirm button when deleting a flight

### PR DESCRIPTION
**Pull request by Prolas**
This dialog box appears when the user wants to delete a flight and asks for confirmation
Added an alert dialog box in flight details.
Added and modified tests to use the DatabaseSetup.
close #239 

image

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/flightdetail/FlightDetailUiTest.kt [hunk 1]
```diff
@@ -13,243 +11,168 @@ import androidx.compose.ui.test.onNodeWithText
...
 
   @Test
-  fun DeleteButtonIsDisplayed() {
-    composeTestRule.onNodeWithText("Delete").assertIsDisplayed()
+  fun deleteButtonWorksWhenDismiss() {
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
+    composeTestRule.onNodeWithText("Delete").performClick()
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("AlertDialogDismiss").performClick()
+    composeTestRule.onNodeWithTag("AlertDialog").assertIsNotDisplayed()
```
Comment (RAFInery) [inline app/src/androidTest/java/ch/epfl/skysync/flightdetail/FlightDetailUiTest.kt:68]: maybe check that still on flight detail page and flight has not been deleted

Comment (RAFInery) [inline app/src/androidTest/java/ch/epfl/skysync/flightdetail/FlightDetailUiTest.kt:68]: maybe in the end also check that after the dismiss we're still on flight detail page and that no flight has been deleted

**Review by RAFInery — APPROVED** [review]
LGTM

File: app/src/main/java/ch/epfl/skysync/screens/flightDetail/FlightDetailUi.kt [hunk 2]
```diff
@@ -51,13 +57,25 @@ import ch.epfl.skysync.models.flight.Vehicle
...
+    AlertDialogExample(
+        onDismissRequest = { showDialog = false },
+        onConfirmation = {
+          showDialog = false
+          deleteClick()
+        },
+        dialogTitle = "Delete Flight",
+        dialogText = "Are you sure you want to delete this flight ?",
+        icon = Icons.Default.Info)
+  }
```
[the remark this answers is not in the document]
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/screens/flightDetail/FlightDetailUi.kt:78]: Ok I will remove it

**Pull request merged**